### PR TITLE
Migrate pkg/kubelet/logs to structured logging

### PR DESCRIPTION
**Pull request by krzysiekg**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Migrate pkg/kubelet/logs to structured logging

#### Which issue(s) this PR fixes:

Ref #98976

#### Does this PR introduce a user-facing change?

```release-note
Migrate pkg/kubelet/logs to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig node
/sig instrumentation
/priority important-soon

### CODE REVIEW

File: pkg/kubelet/logs/container_log_manager.go [hunk 1]
```diff
@@ -412,7 +412,8 @@ func (c *containerLogManager) rotateLatestLog(id, log string) error {
 			// This shouldn't happen.
 			// Report an error if this happens, because we will lose original
 			// log.
-			klog.Errorf("Failed to rename rotated log %q back to %q: %v, reopen container log error: %v", rotated, log, renameErr, err)
+			klog.ErrorS(renameErr, "Failed to rename rotated log back to original, after reopen container log error",
+				"path", rotated, "originalPath", log, "reopenErr", err)
```
Comment (ehashman) [inline pkg/kubelet/logs/container_log_manager.go:416]: /cc @serathius 
on how we want to do multiple errors

**Review by ehashman — COMMENTED** [review]
/ok-to-test
/triage accepted

File: pkg/kubelet/logs/container_log_manager.go (hunk 1 above)
Comment (ehashman) [reply pkg/kubelet/logs/container_log_manager.go:416]: Ah, we had this one already: https://github.com/kubernetes/kubernetes/pull/99848#discussion_r588870925

/close

@krzysiekg thank you for this PR! it conflicts with an earlier one for the same file. would you be able to pick up something else from https://github.com/kubernetes/kubernetes/issues/98976#issuecomment-790927982 ?

Comment (krzysiekg) [reply pkg/kubelet/logs/container_log_manager.go:416]: No problem! My bad, I missed this on the list. I'm already working on another package.

**Pull request closed without merging**